orangetreedinner.rb

```diff
@@ -0,0 +1,146 @@
+# Orange Tree
+# group: Alice, Elia, Jeri
+
+class OrangeTree
+	attr_reader :age, :alive, :grove, :height, :orange_count
+
+	def initialize
+		@age = 0
+		@alive = true
+		@height = 0
+		@orange_count = 0
+	end
+
+	# one year passes on one tree- main method call- calls private methods
+	def one_year_passes
+		grow_older
+		if @alive == true
+			grow_taller
+			grow_oranges
+		end
+
+		if @grove
+			@grove.soil_quality
+		end
+	end
+
+	def pick_orange
+		if @orange_count != 0
+			@orange_count -= 1
+			puts "That was a very tasty orange."
+
+		else
+			puts "There are no more oranges to pick this year."
+		end
+	end
+
+	def plant_on(grove_name)
+		grove_name.trees.push(self)
+		@grove = grove_name
+	end
+
+	# put below methods as private since they should all be called within other methods
+	private
+
+	#trees start growing oranges at 3-- grow fewer oranges based on soil quality
+	def grow_oranges
+		if @age >= 3
+			if @grove
+			 	if @grove.soil_quality == "healthy"
+				@orange_count = @age * 3
+				elsif	@grove.soil_quality == "unhealthy"
+					@orange_count = @age
+				elsif @grove.soil_quality == "average"
+					@orange_count = @age * 2
+				end
+			else
+				@orange_count = @age * 3
+			end
+		end
```

Comment:
This is a lot of nested `if` statements. Can you think of how to make it less nested?